doc/user-converting-existing-binary-into-mystikos-app.md

```diff
@@ -0,0 +1,75 @@
+
+#### 1. ldd
+ldd prints the shared library dependencies for your linker on the command line. ldd shows the transitive closure of dependencies. \
+Eg: In this example we are trying to find the dependencies in a test present in the libcxx2 test suite
+```bash
+ $ ldd appdir/app/llvm-project/libcxx/test/libcxx/atomics/atomics.flag/init_bool.pass.cpp.exe
+        linux-vdso.so.1 (0x00007fffafe68000)
+        libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007ffbf2100000)
+        /lib64/ld-linux-x86-64.so.2 (0x00007ffbf26f3000)
+```
+This invokation tells us that this binary depends on these three shared object libraries which can be copied from the current system into the docker container or installed into the container directly as a package. \
+To copy them over, use the statement `COPY /location/of/shared/object /location/in/container/to/put/it`. \
+To install them directly, do it like you would do on that OS anyway. eg: `RUN apt-get update && apt-get install -y package-name` for ubuntu.
+
+#### 2. readelf -d
+readelf shows you exactly what dependencies the app wants, but this is not a transitive closure so you only get to know the immediate dependencies; it considers separate from dynamic dependency. In a nutshell it determines what the app needs from the host system.\
+Eg: In this example we are trying to find the dependencies in a test present in the libcxx2 test suite
+```bash
+$ readelf -d appdir/app/llvm-project/libcxx/test/libcxx/atomics/atomics.flag/init_bool.pass.cpp.exe | grep NEEDED
+ 0x0000000000000001 (NEEDED)             Shared library: [libc.so.6]
+```
+This invokation tells us that we need to include the libc.so.6 shared object library in our docker container in order for it to be able to run our application. This does not include on-system dependencies such as linux-vdso.so.1 which we may not need to include if we are running on the same OS as the host we are judging on.
+
+
+### Copy over the binary
+
+The next step in this process would be to copy over the binary from the host system into the container using the same COPY command explained above. The statement would look like: 
+```
+COPY /location/of/binary/on/host /location/in/container/to/put/it
+```
+
+### Create the container
+
+The final step in this process would be to create the container needed for the appbuilder. Coherently arrange the statements above so that: 
+1. You have your base OS included
+2. Include any dependencies required for your binary
+3. Include any ENV VARs needed for invokation
+4. Copy the binary you want to execute in mystikos from your host system into the container
+
+You can build and run the container app with the following command to make sure it's correct: \
+`docker run $(docker build -q .) /cmd/to/run/binary <args-if-any>`
+
+The process to create the container is simple as defined by docker. You can simply let us handle it from there by invoking - 
```

Comment:
`The process to create the container is simple as defined by docker`
I am not sure what this line is trying to tell, could you rephrase it a bit?

Reply:
Hey, I added a comma here that I think I needed. Thanks for pointing that out!

Also - mentioned the doc in the getting started readme